### PR TITLE
#63 cast players in TrackerEvent.__str__

### PR DESCRIPTION
**Pull request by StoicLoofah**
This addressed issue #63.

Looking into the issue, it appears that python3 handles format strings differently from python2. In this case, both `None` and `not None` values were failing because neither had properly implemented `__format__` necessary for the alignment formatting provided. The easy fix is to cast to a string first, and that can be properly formatted.

I considered adding `__format__` to `Entity`, but I saw there was already something in there and got scared. 

https://github.com/ggtracker/sc2reader/blob/cc364ffa7ed675cf907e1d62b1bb6e9497007401/sc2reader/objects.py#L157

Maybe someone else will get around to it if this same issue comes up in a another place. At least this fix appears to work for now

### CODE REVIEW

**Comment by StoicLoofah** [comment]
@miguelgondu let me know if this is working for you in the general case

**Comment by Gusgus01** [comment]
@StoicLoofah I think I have updated and tested this change in: https://github.com/StoicLoofah/sc2reader/pull/3 which if you accept you can then update this pull request I think?

There was a rebase involved to bring it up to date with upstream to avoid a merge commit, which I haven't had to do before and might not work.

**Comment by Gusgus01** [comment]
https://github.com/StoicLoofah/sc2reader/pull/4

At this point with all the extraneous commits, it might be worth squashing the commits when merging into ggtracker:upstream. Or maybe pulling your branch down and squashing into two commits if you want to keep the "authors" correct. :)

**Comment by StoicLoofah** [comment]
@Gusgus01 thanks for the tip. I'm not too concerned about the exact commit history, so I will lazily merge as is. thanks for helping to validate and test this fix!

File: test_replays/test_all.py [hunk 1]
```diff
@@ -589,12 +589,12 @@ def test_65895(self):
 
     def test_event_print(self):
         replay = sc2reader.load_replay("test_replays/lotv/lotv1.SC2Replay")
-        with StringIO() as capturedOutput:
-            sys.stdout = capturedOutput
-            for event in replay.events:
-                print(event)
-            self.assertIn("PlayerLeaveEvent", capturedOutput.getvalue())
-            sys.stdout = sys.__stdout__
+        sys.stdout = capturedOutput = StringIO()
```
Comment (cclauss) [inline test_replays/test_all.py:592]: The original __with open() as__ form is safer in the face of exceptions...  Was there a reason to move away from that?

Comment (Gusgus01) [reply test_replays/test_all.py:592]: StringIO.StringIO does not implement exit() I believe and can not be used with with() in python2. io.StringIO does and is what is being used in Python3 so could be used with with().

**Pull request merged**